provider/core/src/erased.rs

```diff
@@ -66,10 +66,13 @@ impl ZeroCopyFrom<dyn ErasedDataStruct> for &'static dyn ErasedDataStruct {
 pub struct ErasedDataStructMarker {}
 
 impl DataMarker<'static> for ErasedDataStructMarker {
-    type Yokeable = &'static dyn ErasedDataStruct;
-    type Cart = dyn ErasedDataStruct;
+    type Yokeable = ErasedDataYokeable;
+    type Cart = ErasedDataYokeable;
 }
 
+#[derive(Yokeable)]
+pub struct ErasedDataYokeable(Rc<dyn ErasedDataStruct>);
```

Comment:
Suggestion: consider the name `ErasedDataStructBox`.

Note that this new type is similar in spirit to the existing class `SerdeSeDataStructWrap`.